internal/runtime/runtime.go

```diff
@@ -0,0 +1,162 @@
+	// Try to match on major.minor.
+	if onlyHasMajorMinor(version) {
+		majorMinor := semver.MajorMinor(version)
+		for _, r := range rts {
+			if strings.HasPrefix("v"+r.Version, majorMinor+".") {
+				return r, nil
+			}
+		}
+		return Runtime{}, fmt.Errorf("no matching %s version found for %s", runtimeDisplayNames[name], version)
+	}
+
+	// Try to match on major version only.
+	if onlyHasMajor(version) {
+		major := semver.Major(version)
+		for _, r := range rts {
+			if strings.HasPrefix("v"+r.Version, major+".") {
+				return r, nil
+			}
+		}
+	}
+
+	return Runtime{}, fmt.Errorf("no matching %s version found for %s", runtimeDisplayNames[name], version)
+}
+
+// GetDefault returns the default version for the specified runtime.
+func GetDefault(runtimes []Runtime, name string) (Runtime, error) {
+	rts := filterByName(runtimes, name)
+	for _, r := range rts {
+		if r.Default {
+			return r, nil
+		}
+	}
+
+	return Runtime{}, fmt.Errorf("no default version found for %s", runtimeDisplayNames[name])
+}
+
+func findRuntimeByAlias(runtimes []Runtime, alias string) (Runtime, error) {
+	als := strings.ToLower(alias)
+	for _, r := range runtimes {
+		for _, a := range r.Alias {
+			if als == a {
+				return r, nil
+			}
+		}
+	}
+
+	return Runtime{}, fmt.Errorf("alias %q not found", alias)
+}
+
+func filterByName(runtimes []Runtime, name string) []Runtime {
+	var rts []Runtime
+	for _, r := range runtimes {
+		if r.Name == name {
+			rts = append(rts, r)
+		}
+	}
+	return rts
+}
+
+func onlyHasMajor(version string) bool {
+	return len(strings.Split(version, ".")) == 1
+}
+
+func onlyHasMajorMinor(version string) bool {
+	return len(strings.Split(version, ".")) == 2
+}
+
+// isFullVersion checks if version follows the full semver format of `{major}.{minor}.{patch}`.
+func isFullVersion(version string) bool {
+	return len(strings.Split(version, ".")) == 3
+}
+
+func (r *Runtime) Validate() error {
+	now := time.Now()
+	if now.After(r.RemovalDate) {
+		fmt.Print(r.removalMsg())
+		return fmt.Errorf("unsupported runtime %s(%s)", runtimeDisplayNames[r.Name], r.Version)
+	}
+
+	if now.After(r.EOLDate) {
+		fmt.Print(r.EOLMsg())
+	}
+	return nil
+}
+
+func (r *Runtime) removalMsg() string {
+	return fmt.Sprintf(
+		"%s%s%s%s",
+		color.RedString(fmt.Sprintf("\n\n%s\n", msg.WarningLine)),
+		color.RedString(fmt.Sprintf("\nThe specified %s(%s) is UNSUPPORTED!\n", runtimeDisplayNames[r.Name], r.Version)),
+		color.RedString(fmt.Sprintf("You MUST update your version of %s to a newer one.\n", runtimeDisplayNames[r.Name])),
+		color.RedString(fmt.Sprintf("\n%s\n\n", msg.WarningLine)),
+	)
+}
+
+func (r *Runtime) EOLMsg() string {
+	return fmt.Sprintf(
+		"%s%s%s",
+		color.RedString(fmt.Sprintf("\n\n%s\n", msg.WarningLine)),
+		color.RedString(fmt.Sprintf(
+			"\nThe specified %s(%s) has reached its EOL. Please upgrade to a newer version.\n",
+			runtimeDisplayNames[r.Name],
+			r.Version,
+		)),
+		color.RedString(fmt.Sprintf("\n%s\n\n", msg.WarningLine)),
+	)
+}
```

Comment:
I think we use "more accurate" wording in https://github.com/saucelabs/saucectl/blob/dd4d97db7c8758bdd2b1e62cc878df4c22a7fd08/internal/saucecloud/cloud.go#L893-L893.

Since the logic is pretty much identical, can't these be consolidated? They're all based on the same inputs of name, version, eol and removal dates.